instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcUtils.java

```diff
@@ -24,6 +27,8 @@
   private static final boolean NORMALIZATION_ENABLED = isQueryNormalizationEnabled("jdbc");
 
   private static Field c3poField = null;
+  private static final Cache<String, SqlStatementInfo> sqlToStatementInfoCache =
+      CacheBuilder.newBuilder().maximumSize(5000).build();
```

Comment:
I wonder if we can use our own `WeakCache` instead of guava-cache - right now it's located in the `javaagent-bootstrap` module, but is only used in `javaagent-tooling`. If we moved it to `javaagent-api` we could use it in the instrumentations too.

Reply:
I like this idea. Would it be ok for me to create a follow-up issue to do this work?

Reply:
Sure! 👍 

I think it could work a bit similar to `WeakMap` (interface in `javaagent-api`, implementation in `javaagent-tooling`).

Reply:
I went ahead and added it here, since it was going to be difficult to get past the muzzle issue without tackling it here.